src/query/storage/m3/dynamic_cluster.go

```diff
@@ -60,16 +61,14 @@ type dynamicCluster struct {
 
 // NewDynamicClusters creates an implementation of the Clusters interface
 // supports dynamic updating of cluster namespaces.
-func NewDynamicClusters(_ DynamicClusterOptions) (Clusters, error) {
-	return nil, errors.New("dynamic cluster configuration not yet supported")
+func NewDynamicClusters(opts DynamicClusterOptions) (Clusters, error) {
```

Comment:
FYI, even though this constructor no longer errors, clients are not actually using this implementation unless they delete their namespaces from the static coordinator config (i.e. migrate).